Implement Subscribe flow

MrFlashAccount opened the pull request:
### Pull Request Description

<!--
- Please describe the nature of your PR here, as well as the motivation for it.
- If it fixes an open issue, please mention that issue number here.
-->

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

MrFlashAccount committed:
Add enterprise plan

MrFlashAccount committed:
Add suspence loading

MrFlashAccount committed:
add some fixes for Subscribe Page

MrFlashAccount committed:
new subscribe-success page

MrFlashAccount committed:
draft result component

MrFlashAccount committed:
Create result component

MrFlashAccount committed:
Implement Subscribe flow

MrFlashAccount committed:
Add specify org name dialog

MrFlashAccount committed:
fixes

MrFlashAccount committed:
fix button

MrFlashAccount committed:
remove unused code

MrFlashAccount committed:
add description

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:
```diff
@@ -15,73 +15,204 @@ import SvgMask from '#/components/SvgMask'
-            typeof className === 'function' ? className(values) : className
-          ),
+    <Tag
+      // @ts-expect-error eventhough typescript is complaining about the type of ariaProps, it is actually correct
```
@somebody1234 do you have any ideas on how can we satisfy the typescript here?

somebody1234 replied:
i guess the conventional solution is to just extract the props and children to variables, and then to explicitly create an `<aria.Link` or  `<aria.Button` in a `switch` or conditional (or ternary)

MrFlashAccount committed:
fix form submission

MrFlashAccount committed:
allow mailto: protocol

MrFlashAccount committed:
put subscribe button in topbar

MrFlashAccount committed:
Show error if acc is not enabled

MrFlashAccount committed:
rename tier -> plan

somebody1234 commented on app/ide-desktop/lib/client/start.ts:
```diff
@@ -9,7 +9,7 @@ import * as esbuild from 'esbuild'
 import * as paths from './paths'
 
-// =================
+// =================app/gui2/vite.config.ts
```
uhoh

MrFlashAccount replied:
🙈

somebody1234 commented on app/ide-desktop/lib/common/src/index.js:
```diff
@@ -27,7 +27,7 @@ export const COMPANY_NAME = 'New Byte Order sp. z o.o.'
-    ['Cross-Origin-Embedder-Policy', 'credentialless'],
-    ['Cross-Origin-Opener-Policy', 'same-origin'],
-    ['Cross-Origin-Resource-Policy', 'same-origin'],
+    // ['Cross-Origin-Embedder-Policy', 'credentialless'],
+    // ['Cross-Origin-Opener-Policy', 'same-origin'],
+    // ['Cross-Origin-Resource-Policy', 'same-origin'],
```
big uhoh. are you sure we can't just blacklist specific assets from getting these headers instead?

MrFlashAccount replied:
Not sure. Is there a way to exclude stripe from checking for CORP?

somebody1234 replied:
i guess not unfortunately? i think you might need to whitelist either index.html or the script that imports stripe? so i guess not as simple as i thought. idk how to feel about completely disabling COOP/CORP/COEP, i think consensus among GUI team was to keep it for security reasons even though it's *technically* no longer needed for profiling.

i think (?) COEP: credentialless was supposed to just make it Just Work, but if that didn't do it then i guess it's fine to leave it like this for now, having it working is more important after all...

MrFlashAccount replied:
I think the main reason for turning COEP/COOP/CORP on is to enable SAB (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/SharedArrayBuffer). Does the GUI team use it? 
I think the only solution would be to either proxy all stripe API/requests/static through our webserver or to open GUI inside an iframe with COEP/COOP/CORP enabled. 

UPD: There's also an option to do nothing and wait for Stripe to add support for COEP/COOP/CORP (AFAIK it's on their roadmap, but I can't provide any proof 😄)

somebody1234 replied:
nah, we never used SAB - we only enabled it for the higher precision performance.now()

re: waiting for Stripe - i assume it's super low priority for them, i wouldn't be surprised if it takes them 1 year+ to do it so we probably can't rely on them implementing it on time

MrFlashAccount replied:
Yea... I pinged @farmaazon for his approval

somebody1234 commented on app/ide-desktop/lib/dashboard/src/App.tsx:
```diff
@@ -363,7 +368,7 @@ function AppRouter(props: AppRouterProps) {
   const routes = (
     <router.Routes>
-      <React.Fragment>
+      <>
```
actually do we even need the fragment? (if so could you add a comment explaining why we can't remove it?)

somebody1234 commented on app/ide-desktop/lib/dashboard/src/App.tsx:
```diff
@@ -376,11 +381,34 @@ function AppRouter(props: AppRouterProps) {
+              <router.Route
+                path={appUtils.SUBSCRIBE_PATH}
+                element={
+                  <errorBoundary.ErrorBoundary>
```
might want to turn
```tsx
                <errorBoundary.ErrorBoundary>
                  <React.Suspense fallback={<loader.Loader />}>
```
into a component

MrFlashAccount replied:
Yea I though about this, but found out that we need a wrapper around a `react-router` `<Route />` component, and we'll put `Suspense` and `ErrorBoundary` there (we'll wrap each route into it's own suspense/error boundary)

somebody1234 commented on app/ide-desktop/lib/dashboard/src/layouts/UserBar.tsx:
```diff
@@ -68,26 +69,41 @@ export default function UserBar(props: UserBarProps) {
       {innerProps => (
         <div
-          className="pointer-events-auto flex h-row shrink-0 cursor-default items-center gap-user-bar rounded-full bg-frame px-icons-x pr-profile-picture backdrop-blur-default"
+          className="pointer-events-auto flex h-row shrink-0 cursor-default items-center gap-1 rounded-full bg-frame px-icons-x pr-profile-picture backdrop-blur-default"
```
prefer semantic tailwind values over numeric ones. if you want to change the values you should just change the underlying CSS variable in the `tailwind.css` file directly.

MrFlashAccount replied:
Hmm, the idea of tailwind is using numeric values. I'm not sure using `semantic` values is a great idea at all.

somebody1234 replied:
> the idea of tailwind is using numeric values

not quite. the idea of tailwind is using *predefined* values. semantic values are better here imo because we can change *all* spacings with a particular *semantic* meaning, in just one line. or in other words, the spacing can be determined by a theme

MrFlashAccount replied:
>  spacing can be determined by a theme

I'm not sure this is what we want to achieve and worth spending our time on. 
IMO, 'w-1' is also semantic, because it doesn't specify what `1` is. We can define this using theme.
On the other hand prop line `gap-user-bar` doesn't give much clue on what is the user bar and where is it.

If we want to have `semantic` classnames, then maybe we should use CSS modules or `vanilla-extract`?

Defining a new selector for each component style is annoying and gives no benefits.

Semantic comes from components

MrFlashAccount replied:
Solution: we'll add tw variants lib https://www.tailwind-variants.org/docs/introduction

somebody1234 commented on app/ide-desktop/lib/dashboard/src/layouts/UserBar.tsx:
```diff
@@ -68,26 +69,41 @@ export default function UserBar(props: UserBarProps) {
+          <ariaComponents.Button
+            variant="icon"
+            size="custom"
+            className="mr-1"
```
but why

MrFlashAccount replied:
??

somebody1234 replied:
normally we use gaps to control spacing in flex and grid containers - is there a reason we meed extra margins here?

MrFlashAccount replied:
![CleanShot 2024-05-07 at 12 05 43@2x](https://github.com/enso-org/enso/assets/61194245/6082b1d6-88d0-43d7-9269-332a1e3055f2)

Yeah, extra margin makes sense here because otherwise the gap bw chat and the next button is too narrow.
Gaps are good when we need to put spaces between equal-sized elements. otherwise, it's better to use margins

somebody1234 replied:
i guess... i think in this situation `p-1` between the icon's container and the icon would be preferable to make the elements the same size, but i don't know whether it'd make the left side look weird

MrFlashAccount replied:
Using margins is preferable here.

somebody1234 commented on app/ide-desktop/lib/dashboard/src/layouts/UserBar.tsx:
```diff
@@ -68,26 +69,41 @@ export default function UserBar(props: UserBarProps) {
+            icon={ChatIcon}
+            aria-label={getText('openHelpChat')}
             onPress={() => {
-              setIsHelpChatOpen(!isHelpChatOpen)
```
i prefer keeping `isHelpChatOpen` in case we want to switch to use the user bar on top of chat. but i guess it's easy enough to add back

somebody1234 commented on app/ide-desktop/lib/dashboard/src/layouts/UserBar.tsx:
```diff
@@ -68,26 +69,41 @@ export default function UserBar(props: UserBarProps) {
+            <ariaComponents.Button
+              rounding="full"
+              size="xsmall"
+              variant="tertiary"
```
consistent props order pls

MrFlashAccount replied:
Not sure it makes sense. 

Maybe we want to add https://eslint-plugin-perfectionist.azat.io/rules/sort-jsx-props , but it's definitely outside of the scope of the feature.

somebody1234 replied:
while true, i think it makes sense to at least be consistent with JSX that is 10 lines away (iow: within the same screen)

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/Card.tsx:
```diff
@@ -0,0 +1,76 @@
+  const { getText } = textProvider.useText()
+
+  return (
+    <div className={tw.twMerge('rounded-xl border border-stone-300 p-4', className)}>
```
please use one of our colors instead (e.g. `border-primary/10`)

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/Card.tsx:
```diff
@@ -0,0 +1,76 @@
+
+  return (
+    <div className={tw.twMerge('rounded-xl border border-stone-300 p-4', className)}>
+      <aria.Heading className="text-3xl leading-10" level={2}>
```
prefer adding classes for headings to `tailwind.config.js` to ensure that we use the same styles for headings everywhere

also use one of the already defined `text-*` classes in `tailwind.config.js` to apply the appropriate `leading`

MrFlashAccount replied:
It's better to define a component for that instead of defining a `className`. Period
I've created a dedicated `Text` components in another PR `Billing page`

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/Card.tsx:
```diff
@@ -0,0 +1,76 @@
+
+      {submitButton != null ? <div className="my-4">{submitButton}</div> : null}
+
+      <div className="-mx-4 w-auto border-[0.5px] border-stone-300" />
```
(see above) avoid using tailwind colors

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/Card.tsx:
```diff
@@ -0,0 +1,76 @@
+      <aria.Heading className="text-3xl leading-10" level={2}>
+        {getText(title)}
+      </aria.Heading>
+      <aria.Text elementType="p" className="text-[16px] leading-6">
```
avoid custom text sizes, tailwind defines more than enough.

also use one of the already defined `text-*` classes in `tailwind.config.js` to apply the appropriate `leading`

MrFlashAccount replied:
See comment above. We'll refactor this later once we add proper typography component

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/ContactSales.tsx:
```diff
@@ -0,0 +1,34 @@
+import * as React from 'react'
```
missing `@file` jsdoc comment

MrFlashAccount replied:
We're not using this file anymore... removed it :)

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/ContactSales.tsx:
```diff
@@ -0,0 +1,34 @@
+import * as ariaComponents from '#/components/AriaComponents'
+
+/**
+ *
```
blank jsdoc comment

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/SubscribeForm.tsx:
```diff
@@ -0,0 +1,107 @@
+        isDisabled={
+          subscribeMutation.isPending || cardElementState == null || cardElementState.error != null
+        }
+        className="mt-1 px-4 py-[6px] text-sm"
```
`py-1.5` instead of `py-[6px]`

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/SubscribeSuccess.tsx:
```diff
@@ -0,0 +1,63 @@
+        title={getText('subscribeSuccessTitle')}
+        subtitle={getText(
+          'subscribeSuccessSubtitle',
+          `${getText(constants.PLAN_TO_TEXT_ID[plan])}`
```
avoid unnecessary template literal (or explain why it is required in a comment)

MrFlashAccount replied:
I think it's unneeded here

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/SubscribeSuccess.tsx:
```diff
@@ -0,0 +1,63 @@
+ * Check if a string is a valid plan.
+ */
+function checkPlan(plan: string): plan is backend.Plan {
+  return plan in backend.Plan
```
use `backend.isPlan` instead of a custom function.
note that you should be checking against values, not keys, of the enum object

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/const.ts:
weird file name. if it is being imported as `constants`, this file should be named `constants.ts`

somebody1234 commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:
```diff
@@ -15,73 +15,207 @@ import SvgMask from '#/components/SvgMask'
 }
 
+/**
+ *
```
blank docs

somebody1234 commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:
```diff
@@ -15,73 +15,207 @@ import SvgMask from '#/components/SvgMask'
   readonly loading?: boolean
-  readonly variant: 'cancel' | 'delete' | 'icon' | 'submit'
+  /**
+   * The variant of the button
```
docs missing trailing `.`. this applies for all docs in this file

MrFlashAccount replied:
I'd rather not address this topic, and I recommend we avoid bringing it up again in the future as it seems to be a matter of personal preference for you to conclude the conversation.

somebody1234 commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:
```diff
@@ -15,73 +15,207 @@ import SvgMask from '#/components/SvgMask'
+  readonly size?: Size
+  readonly rounding?: Roundings
+  /**
+   * Button takes the full width of its container
```
docs should be full sentences

somebody1234 commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx:
```diff
@@ -15,73 +15,207 @@ import SvgMask from '#/components/SvgMask'
+
 const DEFAULT_CLASSES =
-  'flex cursor-pointer rounded-sm border border-transparent transition-[opacity,outline-offset] duration-200 ease-in-out'
+  'flex whitespace-nowrap cursor-pointer border border-transparent transition-[opacity,outline-offset] duration-150 ease-in-out select-none text-center items-center justify-center'
```
should probably just be `duration` to use the default, although it seems like the default is 100ms

MrFlashAccount replied:
Here's the answer. I want 150 ms

somebody1234 commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx:
```diff
@@ -27,7 +28,7 @@ const MODAL_CLASSES_BY_TYPE = {
 
 const DIALOG_CLASSES_BY_TYPE = {
-  modal: 'w-full max-w-md min-h-[200px] h-[90vh] max-h-[90vh]',
+  modal: 'w-full max-w-md min-h-[100px] h-auto max-h-[90vh]',
```
do we really need `h-auto` here?

MrFlashAccount replied:
Not really. thanks for noticing

somebody1234 commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx:
```diff
@@ -63,21 +69,26 @@ export function Dialog(props: types.DialogProps) {
             )}
 
-            <div className="flex-1 shrink-0">
+            <div className="z-1 flex-1 shrink-0 p-3.5">
```
do we actually need `z-1`?

MrFlashAccount replied:
As far as I remember yes, because otherwise the content will be overlayed with ::before backdrop

somebody1234 replied:
`relative` on the child should fix this. while i'm not against z-indexes, i think if we do end up needing them, we should be using named z-indexes for specific purposes, otherwise we just end up with arbitrary z-indexes as a way to brute-force the correct layering.

MrFlashAccount replied:
I don't like z-indexes. Will try your solution :)

somebody1234 commented on app/ide-desktop/lib/dashboard/src/modals/SetOrganizationNameModal.tsx:
```diff
@@ -0,0 +1,124 @@
+        return null
+      }
+    },
+    staleTime: Infinity,
```
probably a bad idea. default `staleTime` should be fine, if it's infinity there's a much higher risk of getting an old value (which can then be edited and submitted, losing any changes in the meantime)

MrFlashAccount replied:
This dialog displays only when the Org is not specified. We can't remove the name of the organization after setting it up. So I think this shouldn't be a problem. Also, eventually, we will manually invalidate this query after changing the name of the organization.

somebody1234 commented on app/ide-desktop/lib/dashboard/src/modals/SetOrganizationNameModal.tsx:
```diff
@@ -0,0 +1,124 @@
+        <aria.Form
+          onSubmit={e => {
+            e.preventDefault()
+            const name = new FormData(e.currentTarget).get('organization')
```
i would prefer controlled inputs for type safety but i guess? this is fine?

MrFlashAccount replied:
IMO it's fine. the form data 'get' returns 'string' | null, so we always check it for existense

somebody1234 commented on app/ide-desktop/lib/dashboard/src/modals/SetOrganizationNameModal.tsx:
```diff
@@ -0,0 +1,124 @@
+        modalProps={{ isOpen: shouldShowModal }}
+      >
+        <aria.Form
+          onSubmit={e => {
```
prefer `event` over `e`

somebody1234 commented on app/ide-desktop/lib/dashboard/src/modals/SetOrganizationNameModal.tsx:
```diff
@@ -0,0 +1,124 @@
+    session && 'user' in session && session.user?.plan != null ? session.user.plan : null
+
+  const queryClient = reactQuery.useQueryClient()
+  const { data } = reactQuery.useSuspenseQuery({
```
prefer renaming to a more descriptive name

somebody1234 reviewed (commented):
mostly (?) stylistic issues?

idk where the major logic changes in this PR are

MrFlashAccount committed:
fix dialog

somebody1234 commented on app/ide-desktop/lib/dashboard/src/modals/SetOrganizationNameModal.tsx:
```diff
@@ -0,0 +1,124 @@
+                clsx('rounded-md border border-gray-300 p-1.5 text-sm transition-[outline]', {
+                  // eslint-disable-next-line @typescript-eslint/naming-convention
+                  'outline outline-2 outline-primary':
+                    values.isFocused || values.isHovered || values.isFocusVisible,
```
wait why on `isFocused` and `isHovered` as well?

MrFlashAccount replied:
Why not?

somebody1234 replied:
we currently use `outline` for focus rings for `:focus-visible` (aka keyboard focus), so this is inconsistent with every other input we have

MrFlashAccount replied:
but it's incorrect from the UX perspective. Inputs should have an outline when they're focused.
But not sure about hover though.

MrFlashAccount replied:
Will remove hover state

somebody1234 replied:
> Inputs should have an outline when they're focused

not sure this is true. just checked and none of google, bing, duckduckgo, gmail and facebook have a focus ring on mouse focus.

although it's true that some places do (the browser URL bar, twitter, github), given that the majority of websites i looked at lack focus rings i assume it's fine to omit it for mouse focus.

MrFlashAccount replied:
> have a focus ring on mouse focus.

What's the mouse focus? Do you mean when a user clicks on the input?

somebody1234 commented on app/ide-desktop/lib/client/src/bin/server.ts:
```diff
@@ -107,10 +108,28 @@ export class Server {
+                organization: 'Hello CA',
+                countryCode: 'NP',
+                state: 'Bagmati',
+                locality: 'Kathmandu',
```
not sure we want dummy values like these, even if *almost* nobody is going to see them. maybe it's better to use the location of the US company - see the bottom of <https://enso.org/> for the address.

the same applies to the organization, not just the location

somebody1234 commented on app/ide-desktop/lib/dashboard/src/pages/subscribe/Subscribe/components/SubscribeForm.tsx:
```diff
@@ -0,0 +1,107 @@
+        throw new Error('Unexpected error')
+      } else {
+        return stripe.createPaymentMethod({ type: 'card', card: cardElement }).then(result => {
+          console.log('Payment method:', result)
```
stray log

somebody1234 commented:
QA :x:
- :x: No way to cancel out of the upgrade flow
- :white_check_mark: `mailto:` for upgrading to "Organization" tier works:
  - :white_check_mark: works in browser
  - :white_check_mark: going to the previous page in the browser (if the mail client is in the same browser) returns the user to the subscribe flow
  - :white_check_mark: works in electron (opens in mail client)
- :warning: Payment flow works
  - Could not test, getting error "Session not complete, please contact the support team or try with another payment method."
  - JSON response from `payments/checkout-sessions/${session_id}` is: `{"status":"open","paymentStatus":"unpaid"}`
- :warning: Electron displays logs in the terminal:
```
----- Certificate i=0 (CN=127.0.0.1) -----
ERROR: No matching issuer found
```
```
electron: Failed to load URL: https://b.stripecdn.com/stripethirdparty-srv/assets/v20.25/HCaptchaInvisible.html?id=8fb50dd4-47e4-4823-b199-d1c439e85ce6&origin=https%3A%2F%2Fjs.stripe.com with error: ERR_BLOCKED_BY_RESPONSE
```
- :white_check_mark: Local Backend in electron is not broken by the HTTP -> HTTPS change
- :warning: `npm run watch:linux` (and counterparts for other OSes) are broken by the HTTP -> HTTPS change


Visual/UX issues:
- "Upgrade" page starts getting cut off at 970px width (and below)
- :warning: "Learn more" button isn't clearly displayed as a link
- "Enso Pricing" header should probably be centered
- "Subscribe" button in "upgrade" modal:
  - isn't `rounded-full`
  - should have height=32px
- In general the new UI has borders which are not present in our design system
  - Also applies to font sizes and line heights
- :warning: As mentioned before, the "close" button for dialogs should be on the left
- The "Subscribe" buttons have a lot of horizontal padding which is also seen nowhere else in our design
- Only the "Subscribe" button in the center changes opacity on hover

MrFlashAccount committed:
fix typo

MrFlashAccount committed:
Fix review issues

MrFlashAccount commented:
@somebody1234 
> Could not test, getting error "Session not complete, please contact the support team or try with another payment method."

known problem on the backend side... Guess it's due to using test bank cards...

> ERROR: No matching issuer found

I think it's fine because we're issuing a self-signed certificate. (we need this to make Stripe live mode work)

> "Upgrade" page starts getting cut off at 970px width (and below)

It's fine because we can scroll...

> "Learn more" button isn't clearly displayed as a link

Agree, it's not clear enough, but it has CTA and hover/focus states... I added Icon to the right and did some styling around this variant... now it should look better

> "Enso Pricing" header should probably be centered

Disagree

> "Subscribe" button in "upgrade" modal:

Disagree

> In general the new UI has borders which are not present in our design system

Could you please share our design system?
Also, as there's no design for the Subscribe page, I implemented it as I like it.

> As mentioned before, the "close" button for dialogs should be on the left

Yeah, I remember. We'll fix this in a separate issue

> The "Subscribe" buttons have a lot of horizontal padding which is also seen nowhere else in our design

They're sized by the width of the container to be large enough to take the attention of the users. It's not an issue.

> Only the "Subscribe" button in the center changes opacity on hover

Thanks for noticing! Fixed

MrFlashAccount committed:
Fix review issues

somebody1234 commented:
> please share our design system?

i believe we don't actually have a dedicated document listing out the design rules. the Figma designs are probably the closest thing we have to an official design spec.

so it's more just an observation that most of our current design lacks borders completely (including modals), and furthermore, apart from inputs, none of our current design uses 1px borders

farmaazon reviewed (commented):
Please check if we need to disable _all_ of these headers (maybe COEP is enough?) and add comment why we're disabling them.

MrFlashAccount commented:
> so it's more just an observation that most of our current design lacks borders completely (including modals), and furthermore, apart from inputs, none of our current design uses 1px borders

Yeah, I agree that the current Figma design doesn't have borders(mostly), but it's too vague to make decisions that we shouldn't have these borders at all.
Also I agree that we need to iterate once more to make the current components even closer to what we have in Figma.

MrFlashAccount committed:
fix load script import

MrFlashAccount committed:
fix dev server work

MrFlashAccount commented:
> ⚠️ npm run watch:linux (and counterparts for other OSes) are broken by the HTTP -> HTTPS change

Fixed.

@somebody1234 ready for re-qa

MrFlashAccount committed:
fix dev server work

MrFlashAccount commented:
@farmaazon looks like removing the COEP header is sufficient. I returned all headers except the COEP.

somebody1234 commented:
QA :white_check_mark:? (with caveats, see below)
- :white_check_mark: can now exit out of subscribe flow
- :white_check_mark: `mailto:` still works in Electron
- :warning: `mailto:` does not open in new tab in browser
- :warning: (expected, but) payment flow seems to still not work due to backend issues?
  - getting 500 errors setting username on `pbuchu` environment
  - as a result, the rest of the flow (e.g.) "set team name" cannot be tested

Nits:
- "Learn More" text does noto look vertically centered next to icon
  - same applies to "Return to Dashboard" text

MrFlashAccount committed:
fix ariaButtonProps

MrFlashAccount committed:
add transitions to tooltips

MrFlashAccount committed:
fix tsc

MrFlashAccount committed:
fix eslint

MrFlashAccount committed:
Fix e2e

farmaazon commented on app/gui2/vite.config.ts:
```diff
@@ -48,7 +48,6 @@ export default defineConfig({
   },
   server: {
     headers: {
-      'Cross-Origin-Embedder-Policy': 'require-corp',
```
Please add a comment here, why this is not used.
